### PR TITLE
Fix: Renaming a run discards its current status and reverts the run status back to UNFINISHED

### LINKED ISSUE #8141: [BUG] Renaming a run discards its current status and reverts the run status back to UNFINISHED
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

- Client: 2.2.2
- Tracking server: 2.2.2

### System information

N/A

### Describe the problem

When renaming a run, the runs status is set to the default status `UNFINISHED`

Observe a run where the status is `FINISHED` with a duration of 36ms:

Screenshot 2023-03-31 at 7 12 45 AM

Click the button to rename the run and change it's name. Notice that it's status has now changed to `UNFINISHED`

Screenshot 2023-03-31 at 7 14 34 AM

The REST API specifies that status may be supplied to https://www.mlflow.org/docs/latest/rest-api.html#update-run. However, that field is optional (which is what I would expect).

Looking at the code, the problem is that the run status is always defaulted when not supplied, and overwrites existing status information in the database.
 
https://github.com/mlflow/mlflow/blob/master/mlflow/store/tracking/sqlalchemy_store.py#L551-L568

### Tracking information

```
System information: Linux #1 SMP Wed Nov 2 17:44:17 UTC 2022
Python version: 3.9.16
MLflow version: 2.2.2
MLflow module location: /opt/conda/lib/python3.9/site-packages/mlflow/__init__.py
Tracking URI: [http://localhost:80](http://localhost/)
Registry URI: [http://localhost:80](http://localhost/)
MLflow environment variables: 
  MLFLOW_TRACKING_TOKEN: {"projectId":"6392bcff9e3a5133532aeae4","runId":"6423013bacb1650c1e6b106e","projectName":"quick-start","environmentId":"63913cf66d2650476805c785","environmentRevisionId":"6422fdcdacb1650c1e6b1061","hardwareTierId":"medium-k8s","mlflowSourceType":"NOTEBOOK","runNumber":33,"datasetInfo":"6392bd049e3a5133532aeae9-6392bd049e3a5133532aeae8"}
  MLFLOW_TRACKING_URI: [http://localhost:80](http://localhost/)
MLflow dependencies: 
  Flask: 1.1.2
  Jinja2: 3.0.3
  alembic: 1.10.2
  click: 8.0.4
  cloudpickle: 2.2.1
  databricks-cli: 0.17.6
  docker: 6.0.1
  entrypoints: 0.4
  gitpython: 3.1.31
  gunicorn: 20.1.0
  importlib-metadata: 6.1.0
  markdown: 3.4.3
  matplotlib: 3.6.2
  numpy: 1.23.3
  packaging: 23.0
  pandas: 1.5.0
  protobuf: 3.20.3
  pyarrow: 8.0.0
  pytz: 2022.2.1
  pyyaml: 6.0
  querystring-parser: 1.2.4
  requests: 2.28.1
  scikit-learn: 1.1.2
  scipy: 1.9.2
  shap: 0.41.0
  sqlalchemy: 1.4.47
  sqlparse: 0.4.3
```

### Code to reproduce issue

Just curling the REST API is sufficient to trigger this bug. I captured the fetch call in my browser session - here's a minimal repro

```
curl 'https://mlflow/ajax-api/2.0/mlflow/runs/update' \
  -H 'Content-Type: application/json; charset=utf-8' \
  --data-raw '{"run_id":"9742e09dafee42f39b9af2bd2e38bf47","run_name":"spiffy-crab-383_renamed"}'
```

### Stack trace

N/A

### Other info / logs

N/A

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [X] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by WeichenXu123**
## Related Issues/PRs

 #8141 

## What changes are proposed in this pull request?

Fix: Renaming a run discards its current status and reverts the run status back to UNFINISHED

Closes #8141 

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
test?

**Comment by WeichenXu123** [comment]
@harupy Sure I will add unit test. Manually tests already passed.

File: mlflow/store/tracking/sqlalchemy_store.py [hunk 1]
```diff
@@ -552,8 +552,10 @@ def update_run_info(self, run_id, run_status, end_time, run_name):
         with self.ManagedSessionMaker() as session:
             run = self._get_run(run_uuid=run_id, session=session)
             self._check_run_is_active(run)
-            run.status = RunStatus.to_string(run_status)
-            run.end_time = end_time
+            if run_status is not None:
```
Comment (dbczumar) [inline mlflow/store/tracking/sqlalchemy_store.py:555]: Does file store handle this correctly?

**Review by dbczumar — APPROVED** [review]
LGTM once we fix file store and add tests. Thanks @WeichenXu123 !

File: mlflow/store/tracking/sqlalchemy_store.py (hunk 1 above)
Comment (WeichenXu123) [reply mlflow/store/tracking/sqlalchemy_store.py:555]: Yes. File store backend invokes https://github.com/mlflow/mlflow/blob/de34d7844a08467ef0339f5441b186fae4adfb78/mlflow/entities/run_info.py#L76 that handle it perfectly.

File: mlflow/server/handlers.py [hunk 2]
```diff
@@ -711,9 +711,11 @@ def _update_run():
         },
     )
     run_id = request_message.run_id or request_message.run_uuid
-    updated_info = _get_tracking_store().update_run_info(
-        run_id, request_message.status, request_message.end_time, request_message.run_name
-    )
+
```
Comment (harupy) [inline mlflow/server/handlers.py:714]:
```suggestion
```

**Comment by harupy** [comment]
Can we also add tests in `tests/tracking/test_rest_tracking.py`?

**Pull request merged**